src/main/java/jenkinsci/plugins/influxdb/generators/SonarQubePointGenerator.java

```diff
@@ -33,13 +33,21 @@
     private static final String SONARQUBE_MINOR_ISSUES = "minor_issues";
     private static final String SONARQUBE_INFO_ISSUES = "info_issues";
     private static final String SONARQUBE_BLOCKER_ISSUES = "blocker_issues";
+    private static final String SONARQUBE_CODE_SMELLS = "code_smells";
+    private static final String SONARQUBE_BUGS = "bugs";
+    private static final String SONARQUBE_COVERAGE = "coverage";
+    private static final String SONARQUBE_VULNERABILITIES = "vulnerabilities";
+    private static final String SONARQUBE_BRANCH_COVERAGE = "branch_coverage";
+    private static final String SONARQUBE_LINE_COVERAGE = "line_coverage";
+    private static final String SONARQUBE_LINES_TO_COVER = "lines_to_cover";
+    private static final String SONARQUBE_DUPLICATED_LINES_DENSITY = "duplicated_lines_density";
 
     private static final String URL_PATTERN_IN_LOGS = ".*" + Pattern.quote("ANALYSIS SUCCESSFUL, you can browse ")
             + "(.*)";
 
     private static final String SONAR_ISSUES_BASE_URL = "/api/issues/search?ps=500&projectKeys=";
 
-    private static final String SONAR_METRICS_BASE_URL = "/api/measures/component?metricKeys=ncloc,complexity,violations&componentKey=";
+    private static final String SONAR_METRICS_BASE_URL = "/api/measures/component?metricKeys=duplicated_lines_density,lines_to_cover,line_coverage,branch_coverage,ncloc,complexity,violations,code_smells,bugs,coverage,vulnerabilities&componentKey=";
```

Comment:
Why are all metric keys added here for retrieval but only the relevant one extracted in `getSonarMetric()`?
I suggest to only include exactly the metric key per request that is required.